idea/idea-frontend-api/src/org/jetbrains/kotlin/idea/frontend/api/types/ktTypeUtils.kt

```diff
@@ -18,4 +29,26 @@ fun KtType.isClassTypeWithClassId(classId: ClassId): Boolean {
 
 private object DefaultTypeClassIds {
     val UNIT = ClassId.topLevel(StandardNames.FqNames.unit.toSafe())
+    val INT = ClassId.topLevel(StandardNames.FqNames._int.toSafe())
+    val LONG = ClassId.topLevel(StandardNames.FqNames._long.toSafe())
+    val SHORT = ClassId.topLevel(StandardNames.FqNames._short.toSafe())
+    val BYTE = ClassId.topLevel(StandardNames.FqNames._byte.toSafe())
+    val FLOAT = ClassId.topLevel(StandardNames.FqNames._float.toSafe())
+    val DOUBLE = ClassId.topLevel(StandardNames.FqNames._double.toSafe())
+    val CHAR = ClassId.topLevel(StandardNames.FqNames._char.toSafe())
+    val BOOLEAN = ClassId.topLevel(StandardNames.FqNames._boolean.toSafe())
+    val STRING = ClassId.topLevel(StandardNames.FqNames.string.toSafe())
 }
+
+val KtType.defaultInitializer: String?
```

Comment:
Make this `internal`?  Perhaps too early to tell, but if UAST-fir is built on top of `KtType` and `UValue` is created atop it, we may need a util that returns default value of the given type. So, I wonder if you can change either the use of `String?` or the util name to indicate the return value is of String format (or both)? E.g.,

```kt
val KtType.defaultInitializer: Any?
    get() = when {
        isMarkedNullable -> null
        ... -> 0
        isBoolean -> false
        ...
    }

internal val KtType.defaultInitializerForQuickFix: String?
    get() = "$defaultInitializer"
```

Reply:
I think `Any` is not a good way to represent the default value. The string format won't work directly either since we need to differentiate `"null"` from `null`, append `f` for floats, and print escaped format of `char`. A better way might be creating an enum for each of the case. But for now I would lean more towards not over engineering for now and do that only when we actaully need it.